community/kernel/src/main/java/org/neo4j/kernel/impl/index/labelscan/NativeLabelScanStore.java

```diff
@@ -290,14 +314,22 @@ public void init() throws IOException
                 monitor.noIndex();
             }
 
-            instantiateTree();
+            isDirty = instantiateTree();
         }
         catch ( MetadataMismatchException e )
         {
             // GBPTree is corrupt. Try to rebuild.
+            isDirty = true;
+        }
+
+        if ( isDirty )
+        {
             monitor.notValidIndex();
             dropStrict();
             instantiateTree();
+        }
+        if ( isDirty || !storeExists )
+        {
```

Comment:
The `|| !storeExists` part is covered by the `needsRebuild = !storeExists;` further up, is it not?